### PR TITLE
fix(core/datetime-picker): responsiveness, strict typing, spacing

### LINKED ISSUE #697: Mobile support for date time picker
### What happened?

Datetimepicker is not responsive for mobile screen sizes

We have to support mobile, tablet and desktop. It would be nice if the date time picker is responsive, currently it is not responsive

### What type of frontend frameware are you seeing the problem on?

Angular

### Which version of iX do you use?

v1.4.0

### Code to produce this issue.

_No response_

### PR DESCRIPTION
**Pull request by matthiashader**
## 💡 What is the current behavior?
- time picker takes not full width on mobile
- spacing is not correct on mobile
- chevron-up / chevron-left is used

GitHub Issue Number: Closes #697, [IX-1614] 

## 🆕 What is the new behavior?

- time picker takes  full width on mobile
- adapted spacing
- bottom margin from the time-picker = 2rem (This was somewhat challenging to achieve because there was no option to edit them externally. As a solution, they were adapted within the component to reflect the changes in the datetime-picker while preserving the spacing in the component. Because of this some spacings have been adapted in the time-picker)
- chevron-up-small / chevron-left-small is used

## 🏁 Checklist

A pull request can only be merged if all of these conditions are met (where applicable):

- [ ] 🦮 Accessibility (a11y) features were implemented
- [ ] 🗺️ Internationalization (i18n) - no hard coded strings
- [ ] 📲 Responsiveness - components handle viewport changes and content overflow gracefully
- [ ] 📄 Documentation was reviewed/updated (`pnpm run docs`)
- [ ] 🧪 Unit tests were added/updated and pass (`pnpm test`)
- [ ] 📸 Visual regression tests were added/updated and pass ([Guide](https://github.com/siemens/ix/blob/main/CONTRIBUTING.md#visual-regression-testing))
- [ ] 🧐 Static code analysis passes (`pnpm lint`)
- [ ] 🏗️ Successful compilation (`pnpm build`, changes pushed)

### CODE REVIEW

File: .changeset/twenty-mangos-grin.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@siemens/ix': patch
+---
+
+fix(core/date-time-picker): responsiveness and spacing, strict typing
```
Comment (danielleroux) [inline .changeset/twenty-mangos-grin.md:5]:
```suggestion
Improve responsive behaviour of `ix-datetime-picker` on small screens 
```

File: packages/core/src/components/datetime-picker/datetime-picker.scss [hunk 2]
```diff
@@ -12,17 +12,14 @@
 
 :host {
   @include ix-component;
+  --btn-width: 100%;
```
Comment (danielleroux) [inline packages/core/src/components/datetime-picker/datetime-picker.scss:15]: What is the idea of using custom css property here? This should not be a setting which a user should customize.

Use should use a scss variable if it used internal.

File: packages/core/src/components/datetime-picker/datetime-picker.scss [hunk 3]
```diff
@@ -35,23 +32,29 @@
 
   @media (min-width: 576px) {
     min-width: max-content;
+    --btn-width: min-content;
 
-    .btn-select-date {
-      left: unset !important;
+    .btn-select-date-container {
+      position: absolute;
```
Comment (danielleroux) [inline packages/core/src/components/datetime-picker/datetime-picker.scss:38]: Do we really need a absolute position here? Looks like the data container can also be placed using `relative` position.

File: packages/core/src/components/datetime-picker/datetime-picker.tsx [hunk 4]
```diff
@@ -87,43 +87,43 @@ export class DatetimePicker
...
+  @Prop() time?: string;
 
   /**
    * Show time reference input
    * Time reference is default aligned with @see {this.timeFormat}
    *
    * @since 1.1.0
    */
-  @Prop() showTimeReference = undefined;
+  @Prop() showTimeReference?: boolean;
```
Comment (danielleroux) [inline packages/core/src/components/datetime-picker/datetime-picker.tsx:113]:
```suggestion
  @Prop() showTimeReference: boolean = false;
```

File: packages/core/src/components/datetime-picker/datetime-picker.tsx [hunk 5]
```diff
@@ -146,7 +146,7 @@ export class DatetimePicker
    *
    * @since 2.1.0
    */
-  @Prop() locale: string = undefined;
+  @Prop() locale?: string = undefined;
```
Comment (danielleroux) [inline packages/core/src/components/datetime-picker/datetime-picker.tsx:149]:
```suggestion
  @Prop() locale?: string;
```

**Review by danielleroux — CHANGES_REQUESTED** [review]

File: packages/core/src/components/date-picker/date-picker.tsx
Comment (nuke-ellington) [inline packages/core/src/components/date-picker/date-picker.tsx:1]: Use updated icon imports

File: packages/core/src/components/date-dropdown/date-dropdown.scss [hunk 6]
```diff
@@ -28,6 +28,7 @@
   }
 
   ix-button {
+    margin-top: 1rem;
```
Comment (nuke-ellington) [inline packages/core/src/components/date-dropdown/date-dropdown.scss]: Why is there a `margin-top` on the button now?

File: packages/core/src/tests/textarea/textarea.e2e.ts-snapshots/textarea-basic-1-chromium---theme-classic-dark-linux.png
Comment (nuke-ellington) [inline packages/core/src/tests/textarea/textarea.e2e.ts-snapshots/textarea-basic-1-chromium---theme-classic-dark-linux.png:1]: Why is the text area affected by this PR?

**Review by nuke-ellington — CHANGES_REQUESTED** [review]

**Pull request merged**